src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -1,60 +1,144 @@
+            isCorrectFormat = false;
+        }
     }
 
     public CommandResult execute() {
+        if (isCorrectFormat) {
+            if (isDeleteAll) {
+                if (confirmDeleteAll()) {
+                    deleteAllEventsAndTasks();
+                    return new CommandResult("I have deleted everything!");
+                } else {
+                    return new CommandResult("I will not delete anything!");
+                }
+            }
 
-        fillCombinedItemList();
+            String deletedItem;
+            if (isEventFlag(itemFlag)) {
+                assert indexToDelete < eventCatalog.size();
+                deletedItem = deleteEvent(indexToDelete);
+                return new CommandResult("I have deleted this event: " + deletedItem);
+            } else if (isTaskFlag(itemFlag)) {
+                assert indexToDelete < eventCatalog.size();
+                deletedItem = deleteTask(indexToDelete);
+                return new CommandResult("I have deleted this task: " + deletedItem);
+            }
+        }
+        return new CommandResult("Unable to delete item!");
+    }
 
-        String removedTaskTitle;
-        removedTaskTitle = deleteTask(taskToDelete);
-        return new CommandResult(Ui.getTaskDeletionMessage(removedTaskTitle));
+    private void prepareInputs(String[] command) throws DukeException {
+        if (command.length == 2) {
+            throw new DukeException("Please give me the index of the event you wish to delete!");
+        } else if (isEventFlag(itemFlag) || isTaskFlag(itemFlag)) {
+            indexToDelete = getIndex(command[2]);
+        } else {
+            throw new DukeException("Invalid item index!");
+        }
+
+        if (!isValidIndex(indexToDelete)) {
+            throw new DukeException("No such item index exists!");
+        }
     }
 
-    public static void fillCombinedItemList() {
-        combinedItemList.clear();
-        combinedItemList.addAll(eventCatalog);
-        combinedItemList.addAll(taskList);
+    private static String deleteEvent(int index) {
+        String eventTitle = eventCatalog.get(index).getTitle();
+        eventCatalog.remove(index);
+        return eventTitle;
     }
 
-    // did not throw a custom exception if taskIndex > itemList.size(), already caught by ArrayIndexOutOfBoundsException
-    public static String deleteTask(String input) {
-        try {
-            int taskIndex = getTaskIndex(input);
-            String taskTitle = combinedItemList.get(taskIndex).getTitle();
-            combinedItemList.remove(taskIndex);
-            return taskTitle;
-        } catch (ArrayIndexOutOfBoundsException e) {
-            return "That task does not exist";
-        } catch (IndexOutOfBoundsException | NumberFormatException e) {
-            return "Please tell me which task number to delete";
-        }
+    private static String deleteTask(int index) {
+        String taskTitle = eventCatalog.get(index).getTitle();
```

Comment:
Should this be `String taskTitle = eventCatalog.get(eventIndex).taskList.get(index).getTitle()` instead? It seems that this line retrieves the title of the current event instead of the desired task.